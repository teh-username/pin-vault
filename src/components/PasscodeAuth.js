import React from 'react';
import { View, StyleSheet } from 'react-native';
import { Button, FormValidationMessage } from 'react-native-elements';

import FormRow from './FormRow';

class PasscodeAuth extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      passcode: '',
    };

    this.handleInputChange = this.handleInputChange.bind(this);
    this.handleFormSubmit = this.handleFormSubmit.bind(this);
  }

  handleFormSubmit() {
    this.props.handlePasscodeSubmit(this.state.passcode);
  }

  handleInputChange(stateKey, data) {
    this.setState({
      [stateKey]: data,
    });
  }

  render() {
    const { error } = this.props;
    const { passcode } = this.state;
    return (
      <View>
        <FormRow
          label="Passcode"
          name="passcode"
          value={passcode}
          onInputChange={this.handleInputChange}
        />
        {error ? <FormValidationMessage>{error}</FormValidationMessage> : null}
        <Button
          title="Enter Passcode"
          icon={{ name: 'lock-open' }}
          onPress={this.handleFormSubmit}
          buttonStyle={styles.buttonStyle}
          disabled={!passcode}
        />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  buttonStyle: {
    top: 40,
    backgroundColor: '#669AE0',
  },
});

export default PasscodeAuth;
