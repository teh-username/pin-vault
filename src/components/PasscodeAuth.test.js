import React from 'react';
import { FormValidationMessage } from 'react-native-elements';

import PasscodeAuth from './PasscodeAuth';

describe('PasscodeAuth component test', () => {
  it('should render as expected', () => {
    const wrapper = shallow(<PasscodeAuth />);
    expect(wrapper.find(FormValidationMessage).length).toEqual(0);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render error message if there are any', () => {
    const wrapper = shallow(<PasscodeAuth error="needs triage" />);
    expect(wrapper.find(FormValidationMessage).prop('children')).toEqual(
      'needs triage'
    );
    expect(wrapper).toMatchSnapshot();
  });
});
