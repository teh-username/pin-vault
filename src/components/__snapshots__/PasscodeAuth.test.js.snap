// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasscodeAuth component test should render as expected 1`] = `
<View>
  <FormRow
    label="Passcode"
    name="passcode"
    onInputChange={[Function]}
    value=""
  />
  <Button
    buttonStyle={
      Object {
        "backgroundColor": "#669AE0",
        "top": 40,
      }
    }
    disabled={true}
    icon={
      Object {
        "name": "lock-open",
      }
    }
    onPress={[Function]}
    title="Enter Passcode"
  />
</View>
`;

exports[`PasscodeAuth component test should render error message if there are any 1`] = `
<View>
  <FormRow
    label="Passcode"
    name="passcode"
    onInputChange={[Function]}
    value=""
  />
  <FormValidationMessage>
    needs triage
  </FormValidationMessage>
  <Button
    buttonStyle={
      Object {
        "backgroundColor": "#669AE0",
        "top": 40,
      }
    }
    disabled={true}
    icon={
      Object {
        "name": "lock-open",
      }
    }
    onPress={[Function]}
    title="Enter Passcode"
  />
</View>
`;
